docs/agent.md

```diff
@@ -87,16 +87,65 @@ $agent = new \Nipwaayoni\Agent(
 
 ## Extending the Agent Class
 
-You can add your own agent functionality by extending the `Agent` class. Note that the `Agent::__construct()` method is declared final and you therefore cannot change the constructor signature when extending the class. The `AgentBuilder` must be able to create new agents with a fixed constructor signature.
+You can add your own agent functionality by extending the `Agent` class.
 
 When you extend the `Agent` class, you should override the `NAME` and `VERSION` class constants as appropriate for your class.
 
 You can also override the `Agent::initialize()` method to execute any object setup. This method will be called by the `Agent::__construct()` method. It is not necessary to call `parent::initialize()`.
 
-Use the `AgentBuilder::withAgentClass()` method to have the builder user your class. For example:
+If you do not change the constructor signature, you can use the `AgentBuilder::withAgentClass()` method to have the builder user your class. For example:
```

Comment:
Actually, two ways to create custom agents can cause confusion to developers. What do you think to keep only AgentBuilder overwritten approach?

Reply:
That's a good point. There is very little additional work needed to extend the `AgentBuilder` class instead of `withAgentClass`. The added value of a "one way to do it" wins. I've removed that in the latest commit.